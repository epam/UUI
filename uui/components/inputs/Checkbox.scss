@use  '../../assets/styles/index.scss' as *;

@mixin icon-sizes($size, $font-size, $line-height) {
    :global(.uui-checkbox) {
        width: $size;
        height: $size;

        :global(.uui-icon) {
            svg {
                width: $size;
            }
        }
    }

    :global(.uui-input-label) {
        font-size: $font-size;
        line-height: $line-height;
    }
}

.root {
    //Colors
    &:hover, &:active, &:focus-within {
        :global(.uui-checkbox) {
            border-color: var(--uui-checkbox-border-hover);
        }

        :global(.uui-checked) {
            background-color: var(--uui-checkbox-bg-checked-hover);
        }
    }

    :global(.uui-checkbox) {
        background-color: var(--uui-checkbox-bg);
        border: 1px solid var(--uui-checkbox-border);
        border-radius: var(--uui-checkbox-border-radius);
    }

    :global(.uui-input-label) {
        // @include typography-inline(); TODO: ???
        font-family: var(--font-regular);
        color: var(--uui-checkbox-label);
        margin-left: 12px;
        flex-grow: 1;
    }

    :global(.uui-checked) {
        background-color: var(--uui-checkbox-bg-checked);
        border-color: var(--uui-checkbox-border-checked);
        fill: var(--uui-checkbox-fill);
    }

    &:global(.uui-disabled),
    &:global(.uui-readonly) {
        cursor: default;

        :global(.uui-checkbox) {
            cursor: default;
            background-color: var(--uui-checkbox-bg-disabled);
            border-color: var(--uui-checkbox-border-disabled);
        }

        &:hover, &:global(.uui-checked), &:focus-within {
            &:global(.uui-checkbox) {
                border-color: var(--uui-checkbox-border-checked-disabled);
            }

            &:global(.uui-icon) {
                background-color: var(--uui-checkbox-bg-checked-disabled);
                border-color: var(--uui-checkbox-border-checked-disabled);
            }
        }

        :global(.uui-input-label) {
            color: var(--uui-checkbox-label-disabled);
        }

        :global(.uui-checked) {
            background-color: var(--uui-checkbox-bg-checked-disabled);
            border-color: var(--uui-checkbox-border-checked-disabled);
        }
    }

    &:global(.uui-invalid) {
        :global(.uui-checkbox) {
            border-color: var(--uui-checkbox-error);
            fill: var(--uui-checkbox-error);

            &:hover, &:active, &:focus-within {
                border-color: var(--uui-checkbox-error);
                fill: var(--uui-checkbox-error);
            }

            &:global(.uui-checked) {
                fill: var(--uui-checkbox-fill);
                background-color: var(--uui-checkbox-error);
                border-color: var(--uui-checkbox-error);

                &:hover, &:active, &:focus-within {
                    background-color: var(--uui-checkbox-error);
                    border-color: var(--uui-checkbox-error);
                }
            }
        }

        :global(.uui-input-label) {
            color: var(--uui-checkbox-error);
        }
    }

    //Sizes
    &.size-18 {
        @include icon-sizes(18px, 14px, 18px);
    }

    &.size-12 {
        @include icon-sizes(12px, 12px, 12px);

        :global(.uui-input-label) {
            margin-left: 6px;
        }
    }

    &.mode-cell {
        justify-content: center;
    }
}