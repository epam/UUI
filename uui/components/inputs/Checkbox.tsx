import { withMods } from '@epam/uui-core';
import * as uuiComponents from '@epam/uui-components';
import { ReactComponent as Check } from '@epam/assets/icons/notification-done-outline.svg';
import { ReactComponent as PartlySelect } from '@epam/assets/icons/content-minus-outline.svg';
import css from './Checkbox.module.scss';

interface CheckboxMods {
    /**
     * Defines component size.
     * @default '18'
     */
    size?: '12' | '18';
    /**
     * Defines the different edit modes.
     * @default 'form'
     */
    mode?: 'form' | 'cell';
}

/** Represents the properties of the Checkbox component. */
export type CheckboxProps = CheckboxMods & uuiComponents.CheckboxProps;

function applyCheckboxMods(mods: CheckboxMods) {
    return [
        css.root,
        css['size-' + (mods.size || '18')],
        css['mode-' + (mods.mode || 'form')],
        'uui-color-primary',
    ];
}

const applyUUICheckboxProps = (props: CheckboxProps) => {
    const defaultIcon = Check;
    const defaultIndeterminateIcon = PartlySelect;
    return {
        icon: props.icon ? props.icon : defaultIcon,
        indeterminateIcon: props.indeterminateIcon ? props.indeterminateIcon : defaultIndeterminateIcon,
    };
};

export const Checkbox = /* @__PURE__ */withMods<uuiComponents.CheckboxProps, CheckboxMods>(uuiComponents.Checkbox, applyCheckboxMods, applyUUICheckboxProps);
