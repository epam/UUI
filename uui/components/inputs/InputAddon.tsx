import { withMods } from '@epam/uui-core';
import { InputAddon as uuiInputAddon, InputAddonProps as uuiInputAddonProps } from '@epam/uui-components';
import css from './InputAddon.module.scss';

function applyInputAddonMods() {
    return [
        css.root,
    ];
}

export const InputAddon = /* @__PURE__ */withMods<uuiInputAddonProps>(uuiInputAddon, applyInputAddonMods);
