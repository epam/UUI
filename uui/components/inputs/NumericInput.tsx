import { devLogger, withMods } from '@epam/uui-core';
import { NumericInput as uuiNumericInput, NumericInputProps as uuiNumericInputProps } from '@epam/uui-components';
import { ControlSize, EditMode, IHasEditMode } from '../types';
import { systemIcons } from '../../icons/icons';
import textInputCss from './TextInput.module.scss';
import css from './NumericInput.module.scss';

const DEFAULT_SIZE = '36';
const DEFAULT_MODE = EditMode.FORM;

type NumericInputMods = IHasEditMode & {
    /**
     * Defines component size
     * @default '36'
     * Size '48' is deprecated and will be removed in future release
     */
    size?: ControlSize;
};

function applyNumericInputMods(mods: NumericInputMods) {
    return [
        textInputCss.root,
        css.root,
        css['size-' + (mods.size || DEFAULT_SIZE)],
        textInputCss['size-' + (mods.size || DEFAULT_SIZE)],
        textInputCss['mode-' + (mods.mode || DEFAULT_MODE)],
    ];
}

/** Represents the properties of a NumericInput component. */
export type NumericInputProps = uuiNumericInputProps & NumericInputMods;

export const NumericInput = /* @__PURE__ */withMods<uuiNumericInputProps, NumericInputMods>(
    uuiNumericInput,
    applyNumericInputMods,
    (props) => {
        if (__DEV__) {
            devLogger.warnAboutDeprecatedPropValue<NumericInputProps, 'size'>({
                component: 'NumericInput',
                propName: 'size',
                propValue: '48',
                propValueUseInstead: '42',
                condition: () => props.size === '48',
            });
        }
        return {
            upIcon: systemIcons.foldingArrow,
            downIcon: systemIcons.foldingArrow,
            align: props.align ?? (props.mode === 'cell' ? 'right' : 'left'),
            disableArrows: props.disableArrows ?? props.mode === 'cell',
        };
    },
);
