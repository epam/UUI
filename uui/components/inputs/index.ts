export * from './Checkbox';
export * from './RadioInput';
export * from './Switch';
export * from './TextInput';
