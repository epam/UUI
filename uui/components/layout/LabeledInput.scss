@use  '../../assets/styles/index' as *;

.root {
    width: 100%;

    :global(.uui-label) {
        @include overflow-ellipsis();
        font-family: var(--font-semibold);
        color: var(--uui-labeled-input);

        :global(.uui-labeled-input-info-icon) {
            flex-shrink: 0;
            margin: 0 4px;
            fill: var(--uui-labeled-input-fill);
        }

        :global(.uui-labeled-input-asterisk) {
            color: var(--uui-labeled-input-asterisk);
        }

        :global(.uui-labeled-input-optional) {
            font-family: var(--font-italic);
            font-size: 12px;
            line-height: 18px;
            color: var(--uui-labeled-input-optional-text);
        }
    }

    :global(.uui-label-left) {
        > :global(.uui-label) {
            margin-right: 5px;
        }
    }

    :global(.uui-invalid-message) {
        line-height: 18px;
        font-size: 12px;
        color: var(--uui-labeled-input-error);
        font-family: var(--font-regular);
        margin-top: 3px;
    }

    // Sizes
    &.size-24 {
        :global(.uui-label-top) {
            > :global(.uui-label) {
                margin-bottom: 6px;
            }
        }

        :global(.uui-label) {
            font-size: 12px;
            line-height: 18px;
        }
    }

    &.size-30 {
        :global(.uui-label-top) {
            > :global(.uui-label) {
                margin-bottom: 6px;
            }
        }

        :global(.uui-label) {
            font-size: 12px;
            line-height: 18px;
        }
    }

    &.size-36 {
        :global(.uui-label-top) {
            > :global(.uui-label) {
                margin-bottom: 6px;
            }
        }

        :global(.uui-label) {
            font-size: 14px;
            line-height: 18px;
        }
    }

    &.size-42 {
        :global(.uui-label-top) {
            > :global(.uui-label) {
                margin-bottom: 6px;
            }
        }

        :global(.uui-label) {
            font-size: 14px;
            line-height: 18px;
        }
    }

    &.size-48 {
        :global(.uui-label-top) {
            > :global(.uui-label) {
                margin-bottom: 6px;
            }
        }

        :global(.uui-label) {
            font-size: 14px;
            line-height: 18px;
        }
    }
}
