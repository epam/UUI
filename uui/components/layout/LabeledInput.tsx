import * as types from '../types';
import { withMods } from '@epam/uui-core';
import * as uuiComponents from '@epam/uui-components';
import { Tooltip } from '../overlays/Tooltip';
import { ReactComponent as FillInfoIcon } from '@epam/assets/icons/notification-info-fill.svg';
import { ReactComponent as InfoIcon } from '@epam/assets/icons/notification-info-outline.svg';
import css from './LabeledInput.module.scss';

const DEFAULT_SIZE = '36';

interface LabeledInputMods extends types.SizeMod {}

/** Represents the properties of the LabeledInput component. */
export type LabeledInputProps = uuiComponents.LabeledInputProps & LabeledInputMods;

function applyLabeledInputMods(mods: LabeledInputMods) {
    return [css.root, css['size-' + (mods.size || DEFAULT_SIZE)]];
}

export const LabeledInput = /* @__PURE__ */withMods<uuiComponents.LabeledInputProps, LabeledInputMods>(uuiComponents.LabeledInput, applyLabeledInputMods, (props) => ({
    Tooltip: props.Tooltip || Tooltip,
    infoIcon: props.infoIcon || (['24', '30'].includes(props.size) ? FillInfoIcon : InfoIcon),
}));
