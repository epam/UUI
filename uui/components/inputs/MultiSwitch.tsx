import * as React from 'react';
import { IEditable, IHasRawProps } from '@epam/uui-core';
import { ButtonProps } from '@epam/uui-components';
import { ControlGroup } from '../layout/ControlGroup';
import { Button, ButtonMods } from '../buttons';
import { SizeMod } from '../types';

interface MultiSwitchItem<TValue> extends ButtonProps, ButtonMods {
    id: TValue;
}

export type UuiMultiSwitchColor = 'primary' | 'secondary';

export interface MultiSwitchProps<TValue> extends IEditable<TValue>, SizeMod, IHasRawProps<React.HTMLAttributes<HTMLDivElement>> {
    items: MultiSwitchItem<TValue>[];
    /**
     * @default 'primary'
     */
    color?: UuiMultiSwitchColor;
}

function MultiSwitchComponent<TValue>(props: MultiSwitchProps<TValue>, ref: React.ForwardedRef<HTMLDivElement>) {
    return (
        <ControlGroup ref={ ref } rawProps={ { ...props.rawProps, role: 'tablist' } }>
            {props.items.map((item, index) => (
                <Button
                    { ...props }
                    { ...item }
                    isDisabled={ props.isDisabled }
                    key={ index + '-' + item.id }
                    onClick={ () => props.onValueChange(item.id) }
                    fill={ props.value === item.id ? 'solid' : 'outline' }
                    color={ props.color }
                    size={ props.size }
                    rawProps={ { 'aria-current': props.value === item.id, role: 'tab' } }
                />
            ))}
        </ControlGroup>
    );
}

export const MultiSwitch = React.forwardRef(MultiSwitchComponent);
