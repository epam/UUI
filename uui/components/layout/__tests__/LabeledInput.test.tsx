import React from 'react';
import { LabeledInput } from '../LabeledInput';
import { TextInput } from '../../inputs';
import { renderSnapshotWithContextAsync } from '@epam/test-utils';

describe('LabeledInput', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <LabeledInput>
                <TextInput value={ null } onValueChange={ () => {} } />
            </LabeledInput>,
        );

        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <LabeledInput
                label='Test label'
                size='36'
                info='Test'
                isInvalid
                validationMessage='Test invalid message'
                labelPosition='left'
            >
                <TextInput value={ null } onValueChange={ () => {} } />
            </LabeledInput>,
        );

        expect(tree).toMatchSnapshot();
    });
});
