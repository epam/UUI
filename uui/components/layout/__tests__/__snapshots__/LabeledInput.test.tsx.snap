// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledInput should be rendered correctly with maximum props 1`] = `
<DocumentFragment>
  <div
    class="container uui-labeled-input root uui-size-30"
  >
    <div
      class="uui-label-left"
    >
      <div
        class="labelWrapper"
      >
        <label
          class="uui-label"
        >
          Test label
          <span
            class="uui-labeled-input-asterisk"
          >
            *
          </span>
          <div
            class="container uui-icon uui-enabled uui-labeled-input-info-icon"
            tabindex="0"
          >
            <svg
              class=""
            />
          </div>
        </label>
        <div
          class="flexSpacer"
        />
        <div
          class="uui-labeled-input-sidenote"
          dir="auto"
        >
          Sidenote text
        </div>
      </div>
      <div
        class="rightChildrenPosition"
      >
        <div
          class="container uui-input-box -clickable root uui-size-36 uui-control-mode-form"
          tabindex="-1"
        >
          <input
            class="uui-input"
            tabindex="0"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="invalidSectionWrapper"
    >
      <div
        class="uui-invalid-message"
        dir="auto"
        role="alert"
      >
        Test validation message
      </div>
      <div
        class="uui-labeled-input-char-counter"
        dir="auto"
      >
        0/10
      </div>
    </div>
    <div
      class="footnoteSectionWrapper"
    >
      <div
        class="uui-labeled-input-footnote"
        dir="auto"
      >
        Footnote text
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LabeledInput should be rendered correctly with minimum props 1`] = `
<DocumentFragment>
  <div
    class="container uui-labeled-input root uui-size-36"
  >
    <div
      class="uui-label-top"
    >
      <div>
        <div
          class="container uui-input-box -clickable root uui-size-36 uui-control-mode-form"
          tabindex="-1"
        >
          <input
            class="uui-input"
            tabindex="0"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
